common/persistence/versionhistory/version_history.go

```diff
@@ -129,29 +129,54 @@ func ContainsVersionHistoryItem(v *historyspb.VersionHistory, item *historyspb.V
 
 // FindLCAVersionHistoryItem returns the lowest common ancestor VersionHistoryItem.
 func FindLCAVersionHistoryItem(v *historyspb.VersionHistory, remote *historyspb.VersionHistory) (*historyspb.VersionHistoryItem, error) {
-	localIndex := len(v.Items) - 1
-	remoteIndex := len(remote.Items) - 1
+	return findLCAVersionHistoryItemFromItemSlice(v.Items, remote.Items)
+}
+
+func findLCAVersionHistoryItemFromItemSlice(a []*historyspb.VersionHistoryItem, b []*historyspb.VersionHistoryItem) (*historyspb.VersionHistoryItem, error) {
+	aIndex := len(a) - 1
+	bIndex := len(b) - 1
 
-	for localIndex >= 0 && remoteIndex >= 0 {
-		localVersionItem := v.Items[localIndex]
-		remoteVersionItem := remote.Items[remoteIndex]
+	for aIndex >= 0 && bIndex >= 0 {
+		aVersionItem := a[aIndex]
+		bVersionItem := b[bIndex]
 
-		if localVersionItem.Version == remoteVersionItem.Version {
-			if localVersionItem.GetEventId() > remoteVersionItem.GetEventId() {
-				return CopyVersionHistoryItem(remoteVersionItem), nil
+		if aVersionItem.Version == bVersionItem.Version {
+			if aVersionItem.GetEventId() > bVersionItem.GetEventId() {
+				return CopyVersionHistoryItem(bVersionItem), nil
 			}
-			return localVersionItem, nil
-		} else if localVersionItem.Version > remoteVersionItem.Version {
-			localIndex--
+			return aVersionItem, nil
+		} else if aVersionItem.Version > bVersionItem.Version {
+			aIndex--
 		} else {
-			// localVersionItem.Version < remoteVersionItem.Version
-			remoteIndex--
+			// aVersionItem.Version < bVersionItem.Version
+			bIndex--
 		}
 	}
 
 	return nil, serviceerror.NewInternal("version history is malformed. No joint point found.")
 }
 
+// IsSameBranch checks if two version history items are in the same branch
+func IsVersionItemsInSameBranch(a []*historyspb.VersionHistoryItem, b []*historyspb.VersionHistoryItem) bool {
```

Comment:
this function should take in `*historyspb.VersionHistory` not `[]*historyspb.VersionHistoryItem`

Reply:
Sure. Will add another method which take in *historyspb.VersionHistory.

Reply:
>Sure. Will add another method which take in *historyspb.VersionHistory.

no this is not what i am asking
version contains 2 things
1. compressed info about event ID & version
2. persistence branch token
persistence branch token does not matter in this case (i.e. no divergence between 2 version histories)
 
```go
func IsVersionItemsInSameBranch(
    thisVersionHistory *historyspb.VersionHistory, 
    thatVersionHistory *historyspb.VersionHistory,
) bool {
    // business logic
}
```

Reply:
Yeah, I understand. The problem is that `ExecutableHistoryTask` and 'replicationspb.HistoryTaskAttributes' only has `[]*historyspb.VersionHistoryItem`. Shall we build a new VersionHistory object with empty branch token and then call this method?